cli/src/action/database/mod.rs

```diff
@@ -63,6 +77,227 @@ impl Action for MigrateAction {
+            }
+            ImportError::Proposal(requester, _) => write!(
+                f,
+                "error importing proposal for made by requester {}",
+                requester
+            ),
+            ImportError::ListCircuits(_) => write!(f, "error listing circuits"),
+            ImportError::ListProposals(_) => write!(f, "error listing proposals"),
+        }
+    }
+}
+
+impl Error for ImportError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self {
+            ImportError::Circuit(_, admin_store_error) => Some(admin_store_error),
+            ImportError::Proposal(_, admin_store_error) => Some(admin_store_error),
+            ImportError::ListCircuits(admin_store_error) => Some(admin_store_error),
+            ImportError::ListProposals(admin_store_error) => Some(admin_store_error),
+        }
+    }
+}
+
+struct ImportResult {
+    nodes: usize,
+    proposals: usize,
+}
+
+impl Default for ImportResult {
+    fn default() -> Self {
+        ImportResult {
+            nodes: 0,
+            proposals: 0,
+        }
+    }
+}
+
+fn import_store(
+    to: Box<dyn AdminServiceStore>,
+    from: Box<dyn AdminServiceStore>,
+) -> Result<ImportResult, ImportError> {
+    let mut import_result = ImportResult::default();
+    let predicates = &[];
+    let circuits = from
+        .list_circuits(predicates)
+        .map_err(ImportError::ListCircuits)?;
+
+    for circuit in circuits {
+        let id = circuit.circuit_id().to_string();
+        let members = circuit.members().to_vec().clone();
+        to.add_circuit(circuit, members)
+            .map_err(|e| ImportError::Circuit(id, e))?;
+        import_result.nodes += 1;
+    }
+
+    let proposals = from
+        .list_proposals(predicates)
+        .map_err(ImportError::ListProposals)?;
+    for proposal in proposals {
+        let requester = proposal.requester_node_id().to_string();
+        to.add_proposal(proposal)
+            .map_err(|e| ImportError::Proposal(requester, e))?;
+        import_result.proposals += 1;
+    }
+
+    Ok(import_result)
+}
+
+impl Action for MigrateFromYamlAction {
+    fn run<'a>(&mut self, arg_matches: Option<&ArgMatches<'a>>) -> Result<(), CliError> {
+        let (state_dir, database_uri) = match arg_matches {
+            Some(args) => (
+                args.value_of("state_dir"),
+                match args.value_of("connect") {
+                    Some(url) => url.to_string(),
+                    None => get_default_database()?,
+                },
+            ),
+            None => (None, get_default_database()?),
+        };
+
+        let state_dir = match state_dir {
+            Some(state_dir) => state_dir.to_string(),
+            None => Self::env_state_dir()?,
+        };
+
+        Self::import_command(&mut io::stdout(), &state_dir, || {
+            // Verify database connection before setting up pool
+            PgConnection::establish(&database_uri[..]).map_err(|err| {
```

Comment:
Commit message says that only Postgress is supported, however this command should support sqlite as well.